1.sh

```diff
@@ -1,35 +1,34 @@
 #! /bin/sh
 
-if [-d $1] && [-d $2]
+if [ -d "$1" ] && [ -d "$2" ]
```

Comment:
In shell, spacing is _very_ important. Since `[` is essentially a command (a shell builtin), the spacing rules must apply.

Remember to protect your variables (even your assignments, where necessary) to ensure bugs don't arise in your programs. It's very important. Had I not fixed this, you'd have ran into errors on this line alone, due to the possibility of passing one or two directory names with spaces, or perhaps special characters bash would otherwise interpret.

Reply:
Thanks for your contribution. It seems my common mistake was with "$x", and I aprriciate your correction! I will upload newer version soon, hope you can check it and spot the mistakes.